commerce/domain/shipping/splitshipment/SplitShipmentSample.cls

```diff
@@ -0,0 +1,128 @@
+     */
+    global SplitShipmentSample() {}
+
+    public virtual override void arrangeItems(CartExtension.ItemArrangementRequest request) {
+        CartExtension.Cart cart = request.getCart();
+
+        Iterator<CartExtension.ItemArrange> itemArrangeIterator = request.getItemArrangeList().iterator();
+        while (itemArrangeIterator.hasNext()) {
+            arrangeItemsByCartItemQuantity(cart, itemArrangeIterator.next());
+        }
+    }
+
+    /**
+     * @description Item arrangement algorithm based on a maximum CartItem quantity per CartDeliveryGroup.
+     * @param cart Holds details about a Cart
+     * @param itemArrange Holds details of how to arrange an item between CartDeliveryGroups
+     */
+    private void arrangeItemsByCartItemQuantity(CartExtension.Cart cart, CartExtension.ItemArrange itemArrange) {
+        // Find the CartItem using the cartItemId provided by the ItemArrange object
+        Iterator<CartExtension.CartItem> cartItemsIterator = cart.getCartItems().iterator();
+        while (cartItemsIterator.hasNext()) {
+            CartExtension.CartItem cartItem = cartItemsIterator.next();
+            if (cartItem.getId() == itemArrange.getCartItemId()) {
+                // Check to see if we need to create more CartDeliveryGroups based on quantity
+                if (shouldSplitShipmentByQuantity(cartItem.getQuantity(), itemArrange.getQuantity())) {
+                    splitShipmentByQuantity(cart, cartItem, itemArrange);
+                }
+            }
+        }
+    }
+
+    /**
+     * @description Determines whether or not the shipment should be split into multiple shipments, based on quantity.
+     * @param cartItemQuantity Quantity associated to a CartItem
+     * @param maxCartItemQuantityPerDeliveryGroup Maximum CartItem quantity per CartDeliveryGroup (provided by ItemArrange)
+     *
+     * @return Boolean True if the quantity associated to the CartItem exceeds the quantity associated to ItemArrange
+     */
+    private Boolean shouldSplitShipmentByQuantity(Decimal cartItemQuantity, Decimal maxCartItemQuantityPerDeliveryGroup) {
+        return cartItemQuantity > maxCartItemQuantityPerDeliveryGroup;
+    }
+
+    /**
+     * @description Splits the shipment into N CartItems and N CartDeliveryGroups, based on quantity.
+     * @param cart Holds details about a Cart
+     * @param cartItem Holds details about a CartItem
+     * @param itemArrange Holds details of how to arrange an item between CartDeliveryGroups
+     */
+    private void splitShipmentByQuantity(CartExtension.Cart cart, CartExtension.CartItem cartItem, CartExtension.ItemArrange itemArrange) {
+        // Adjust the original CartItem's quantity to maxCartItemQuantityPerDeliveryGroup and create N number of
+        // CartItems and CartDeliveryGroups until we've accounted for the entire quantity
+        Decimal maxCartItemQuantityPerDeliveryGroup = itemArrange.getQuantity();
```

Comment:
so you are treating the itemArrange.quantity as a quantity limit here?

Reply:
Yes, is that how it's supposed to be used?

Reply:
No, the default implementation use this quantity as the target quantity to change to.